src/effector/fork/createScope.ts

```diff
@@ -127,7 +128,31 @@ export function createScope(unit?: Domain): Scope {
     fxCount: forkInFlightCounter,
     storeChange,
     warnSerializeNode,
-    activeEffects: [],
+    scopeRef: {ref: null as unknown as Scope},
   }
+  // set initial scope ref to the scope itself
+  resultScope.scopeRef.ref = resultScope
+
+  if (is.scope(baseUnit)) {
+    const oldScope = baseUnit
+    // Update old scope scopeRef
+    // to redirect all effect.finally and scope-binded events to new scope
+    oldScope.scopeRef.ref = resultScope
+
+    // transfer old scope scopeRef to the new scope,
+    // so any further fork(scope) calls will also redirect any effect.finally and scope-binded events to new scope
+    resultScope.scopeRef = oldScope.scopeRef
+
+    // transfer old scope state to the new one
+    resultScope.sidValuesMap = Object.assign({}, oldScope.sidValuesMap)
```

Comment:
Usually we use object spread except cases with mutable assignments
```suggestion
    resultScope.sidValuesMap = {...oldScope.sidValuesMap
```